src/backend/mailbox_manager/api/client/viewsets.py

```diff
@@ -293,6 +293,15 @@ def enable(self, request, domain_slug, pk=None):  # pylint: disable=unused-argum
         mailbox.save()
         return Response(serializers.MailboxSerializer(mailbox).data)
 
+    @action(detail=True, methods=["post"])
+    def reset_password(self, request, domain_slug, pk=None):  # pylint: disable=unused-argument
+        """Send a request to dimail to change password
+        and email new password to mailbox's secondary email."""
+        mailbox = self.get_object()
+        client = DimailAPIClient()
+        client.reset_password(mailbox)
+        return Response(serializers.MailboxSerializer(mailbox).data)
```

Comment:
NIT: The mailbox data do not change here, I'm not sure why we return the mailbox data.